diff --git a/examples/physics/FullCMS/Geant4/CMakeLists.txt b/examples/physics/FullCMS/Geant4/CMakeLists.txt
index 1a927501..f83f3361 100644
--- a/examples/physics/FullCMS/Geant4/CMakeLists.txt
+++ b/examples/physics/FullCMS/Geant4/CMakeLists.txt
@@ -15,6 +15,12 @@ find_package(Geant4 REQUIRED)
 #
 include(${Geant4_USE_FILE})
 include_directories(${PROJECT_SOURCE_DIR}/include)
+get_property(TMP_INCLUDE_DIRECTORIES DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY INCLUDE_DIRECTORIES)
+list(REMOVE_ITEM TMP_INCLUDE_DIRECTORIES "${VECGEOM_INCLUDE_DIR}" "${VecCore_INCLUDE_DIR}")
+set_property(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY INCLUDE_DIRECTORIES ${TMP_INCLUDE_DIRECTORIES})
+remove_definitions(-DVECGEOM_VC)
+include_directories(${OldVc_INCLUDE_DIR})
+list(APPEND Geant4_LIBRARIES ${OldVc_INCLUDE_DIR}/../lib/libVc.a)
 
 #----------------------------------------------------------------------------
 # Locate sources and headers for this project
diff --git a/examples/physics/TestEm3/Geant4/CMakeLists.txt b/examples/physics/TestEm3/Geant4/CMakeLists.txt
index d9f9cd84..ac9111e4 100644
--- a/examples/physics/TestEm3/Geant4/CMakeLists.txt
+++ b/examples/physics/TestEm3/Geant4/CMakeLists.txt
@@ -19,6 +19,12 @@ endif()
 # Setup Geant4 include directories and compile definitions
 #
 include(${Geant4_USE_FILE})
+get_property(TMP_INCLUDE_DIRECTORIES DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY INCLUDE_DIRECTORIES)
+list(REMOVE_ITEM TMP_INCLUDE_DIRECTORIES "${VECGEOM_INCLUDE_DIR}" "${VecCore_INCLUDE_DIR}")
+set_property(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY INCLUDE_DIRECTORIES ${TMP_INCLUDE_DIRECTORIES})
+remove_definitions(-DVECGEOM_VC)
+include_directories(${OldVc_INCLUDE_DIR})
+list(APPEND Geant4_LIBRARIES ${OldVc_INCLUDE_DIR}/../lib/libVc.a)
 
 #----------------------------------------------------------------------------
 # Locate sources and headers for this project
diff --git a/examples/physics/TestEm5/Geant4/CMakeLists.txt b/examples/physics/TestEm5/Geant4/CMakeLists.txt
index 8195938f..671cef97 100644
--- a/examples/physics/TestEm5/Geant4/CMakeLists.txt
+++ b/examples/physics/TestEm5/Geant4/CMakeLists.txt
@@ -19,6 +19,12 @@ endif()
 # Setup Geant4 include directories and compile definitions
 #
 include(${Geant4_USE_FILE})
+get_property(TMP_INCLUDE_DIRECTORIES DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY INCLUDE_DIRECTORIES)
+list(REMOVE_ITEM TMP_INCLUDE_DIRECTORIES "${VECGEOM_INCLUDE_DIR}" "${VecCore_INCLUDE_DIR}")
+set_property(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY INCLUDE_DIRECTORIES ${TMP_INCLUDE_DIRECTORIES})
+remove_definitions(-DVECGEOM_VC)
+include_directories(${OldVc_INCLUDE_DIR})
+list(APPEND Geant4_LIBRARIES ${OldVc_INCLUDE_DIR}/../lib/libVc.a)
 
 #----------------------------------------------------------------------------
 # Locate sources and headers for this project
