diff --git a/magneticfield/inc/Geant/SimpleIntegrationDriver.h b/magneticfield/inc/Geant/SimpleIntegrationDriver.h
index e6a39e2a..1c8089a7 100644
--- a/magneticfield/inc/Geant/SimpleIntegrationDriver.h
+++ b/magneticfield/inc/Geant/SimpleIntegrationDriver.h
@@ -659,7 +659,7 @@ void SimpleIntegrationDriver<T_Stepper, Nvar>::OneGoodStep(const Real_v yStart[]
 
   // vecCore::Int_v  finishedInt = 0;
 
-  Real_v hFinal = htry, errmax_sqFinal; // xFinal, hdidFinal,
+  Real_v hFinal = htry, errmax_sqFinal = 0.0; // xFinal, hdidFinal,
   // Real_v yFinal[ncompSVEC];
   Bool_v goodStep(false), stepSizeUnderflow(false);
 
